src/shared/test/download-file-test.js

```diff
@@ -18,26 +18,46 @@ describe('download-json-file', () => {
         removeChild: sinon.stub(),
       },
     };
-  });
 
-  it('generates export file with provided annotations', () => {
-    const filename = 'my-file.json';
-    const data = { foo: ['bar', 'baz'] };
-
-    const fileContent = downloadJSONFile(data, filename, fakeDocument);
+    sinon.spy(window, 'Blob');
+  });
 
-    assert.equal(fileContent, JSON.stringify(data, null, 2));
+  afterEach(() => {
+    window.Blob.restore();
```

Comment:
`URL.createObjectURL` does not return anything useful to assert against, as it's just a temporary URL that the browser knows how to handle, but does not expose any information about the content it points to.

That's why I have decided to spy on `Blob`. That way we can verify the right content and type is behind that temporary URL.

Reply:
You can get the content of the text file by `fetch`-ing the URL and then reading the text body. The upside is this avoids mocking internals, the downside is that it requires the use of async.

Reply:
Hey, that's not a bad idea.